docs/BUILD.bazel

```diff
@@ -196,3 +196,114 @@ sh_binary(
     data = ["merge_" + k for k in _DOCS],
     target_compatible_with = _TARGET_COMPATIBLE_WITH,
 )
+
+####
+
+# Copyright 2023 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Generated documentation for the docs/ directory
+
+load(":docgen.bzl", "sphinx_stardocs")
+
+sphinx_stardocs(
```

Comment:
IMHO this could be a separate workspace within this `git` repo. Since Sphinx is a Python project, having `sphinxdocs` next to `python` at the root of the repo would be an interesting way to share code with others.

Reply:
fwiw we (envoy) have quite a bit of code wrapping bazel/sphinx, which we reuse in a few places

i had been thinking to factor out related code, push more to bazel and shift to our tooling repo - i think a bazel sphinx rule would be generally useful (putting aside ongoing maintenance issue wrt sphinx and bazel support specifically)

Reply:
Yeah, it's definitely the sort of functionality that should be more generally usable. Putting it in the stardoc rules was floated as an idea, too, when I initially did this for rules_testing. I wrote the code to be mostly re-usable, but I think it will need a couple more iterations to entirely tease out how to make it a stand-alone project. There are various issues like

I moved the sphinx code to //sphinxdocs in that vein to better conceptually separate rules_python's docs from sphinx stuff. I don't want to create a separate workspace, though. Partially because we don't have sufficient CI slots to test it, but also because I think it's premature to do so (the stardoc templates, theme, css, myst extension, and markdown design decisions are all somewhat intertwined, and each are project-specific decisions; untangling that is more work than I can put in right now), and it involves having to wire more things together, which just seems like extra work with little gain at this point.